framework/src/org/checkerframework/framework/test/DefaultCheckerTest.java

```diff
@@ -0,0 +1,88 @@
+package org.checkerframework.framework.test;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.annotation.processing.AbstractProcessor;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.checkerframework.framework.test.TestConfigurationBuilder.buildDefaultConfiguration;
+
+/**
+ * To use this class you must do two things:
+ * 1) Create exactly 1 constructor in the subclass with exactly 1 argument which is a java.io.File.  This File
+ * will be the Java file that is compiled and whose output is verified.
+ * 2) Create one of the following 2 public static methods with the annotation with org.junit.runners.Parameterized.Parameters.
+ * The method name and signature must match exactly (though you can place @Parameters on the line above the method).
+ *
+ * {@code @Parameters public static String [] getTestDirs()}
+ * getTestDir must return an array of directories that exist in the test folder, e.g.
+ * {@code @Parameters public static String [] getTestDirs() { return new String[]{"all-systems", "flow"}}}
+ * The directories can contain more path information (e.g.  "myTestDir/moreTests") but note,
+ * the testsuite will find all of the java test files that exists below the listed directories.
+ * It is unnecessary to list child directories of a directory you have already listed.
+ *
+ * {@code @Parameters public static List<String> getTestFiles(); }
+ * The method returns a List of Java files. There are methods in TestUtilities like findNestedJavaTestFiles to help
+ * you construct this List. The TestSuite will then instantiate the subclass once for each file returned by
+ * getTestFiles and execute the run method.
+ * An example of this method is:
+ *
+ * {@code @Parameters public static List<Object[]> getTestFiles() { return TestUtilities.findNestedJavaTestFiles("aggregate"); } }
+ *
+ */
+@RunWith(TestSuite.class)
+public abstract class DefaultCheckerTest {
```

Comment:
I would still call it CheckerFrameworkTest, because it's not a default implementation for some interface.
But this is OK. (I saw a comment from you on this in email, but not here...)